src/main/java/com/example/demo/Service/StudentService.java

```diff
@@ -19,10 +19,32 @@ public StudentService(){
     }
 
 
+
     public List<Student> getAllStudents() {
+
         return students;
     }
 
+    public Student getStudentById(Long id) {
```

Comment:
This is a perfect case for **Optional<Student>**.

Best practices are that you always return an **Optional** if you methods can return **nulls**.

This way it will force the caller to handle null cases at compile time.
Once you change this to Optional, you will see what changes you need to make to your code to further compile and check for Optional presence.